### DISCUSSION TITLE
Error in mx.nd.internal.empty.array(shape, ctx) :    [14:58:38] C:\Jenkins\workspace\mxnet\mxnet\src\storage\storage.cc:137: Compile with USE_CUDA=1 to enable GPU usage in R

### DISCUSSION THREAD

**Discussion by jasperDD** [apache/incubator-mxnet#14104] [Q&A]
I run this script for captcha recognition in R (windows machine)
mx.set.seed(42)

model <- mx.model.FeedForward.create(
  X                  = train,
  eval.data          = val,
  ctx                = mx.gpu(),
  symbol             = captcha_net,
  eval.metric        = mx.metric.acc2,
  num.round          = 10,
  learning.rate      = 0.0001,
  momentum           = 0.9,
  wd                 = 0.00001,
  batch.end.callback = mx.callback.log.train.metric(50),
  initializer        = mx.init.Xavier(factor_type = "in", magnitude = 2.34),
  optimizer          = "sgd",
  clip_gradient = 10
)

Then, it returns error
[14:58:38] c:\jenkins\workspace\mxnet\mxnet\src\operator\tensor\./matrix_op-inl.h:166: Using target_shape will be deprecated.
Start training with 1 devices
[14:58:38] c:\jenkins\workspace\mxnet\mxnet\src\operator\tensor\./matrix_op-inl.h:166: Using target_shape will be deprecated.
Error in mx.nd.internal.empty.array(shape, ctx) : 
  [14:58:38] C:\Jenkins\workspace\mxnet\mxnet\src\storage\storage.cc:137: Compile with USE_CUDA=1 to enable GPU usage
> 

What;s wrong and how to fix it?

**Comment by jasperDD** [comment]
I understood what the problem. I don't have Nvidia card, so i must use MXnet only CPU and not GPU
like this
model <- mx.model.FeedForward.create(
  X                  = train,
  eval.data          = val,
  ctx                = **mx.cpu()**,
  symbol             = captcha_net,
  eval.metric        = mx.metric.acc2,
  num.round          = 10,
  learning.rate      = 0.0001,
  momentum           = 0.9,
  wd                 = 0.00001,
  batch.end.callback = mx.callback.log.train.metric(50),
  initializer        = mx.init.Xavier(factor_type = "in", magnitude = 2.34),
  optimizer          = "sgd",
  clip_gradient = 10
)

**Comment by lanking520** [comment]
@jasperDD Thanks for identifying the problem. Can you confirm that the model is using GPU as a default?
@anirudhacharya @hetong007

**Comment by Ishitori** [comment]
Yes, @lanking520, it is using GPU by default. See it here: https://github.com/apache/incubator-mxnet/blob/master/example/captcha/mxnet_captcha.R

**Comment by lanking520** [comment]
@Ishitori Thanks for your input. Should we make it cpu as a default?
